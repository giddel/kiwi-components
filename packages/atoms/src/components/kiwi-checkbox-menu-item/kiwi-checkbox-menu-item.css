:host {
  display: block;
}

.checkbox-menu-item .checkbox label {
  display: inline-block;
  width: 100%;
}
