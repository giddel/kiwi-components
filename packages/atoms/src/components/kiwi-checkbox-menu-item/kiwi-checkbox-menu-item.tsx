import { Component, ComponentInterface, h, JSX } from '@stencil/core';

@Component({
  tag: 'kiwi-checkbox-menu-item',
  styleUrl: 'kiwi-checkbox-menu-item.css',
  shadow: false,
})
export class KiwiCheckboxMenuItem implements ComponentInterface {
  render(): JSX.Element {
    return (
      <li class="checkbox-menu-item pointer">
        <slot />
      </li>
    );
  }
}
