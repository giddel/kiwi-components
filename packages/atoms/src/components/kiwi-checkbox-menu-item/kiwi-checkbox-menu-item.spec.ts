import { newSpecPage } from '@stencil/core/testing';
import { KiwiCheckboxMenuItem } from './kiwi-checkbox-menu-item';

describe('kiwi-checkbox-menu-item', () => {
  it('builds', async () => {
    const item = await newSpecPage({
      components: [KiwiCheckboxMenuItem],
      html: `<kiwi-checkbox-menu-item></kiwi-checkbox-menu-item>`,
    });
    expect(item.root).toBeTruthy();
  });

  it('renders', async () => {
    const item = await newSpecPage({
      components: [KiwiCheckboxMenuItem],
      html: `<kiwi-checkbox-menu-item><span>Slot Child</span></kiwi-checkbox-menu-item>`,
    });

    expect(item.root).toMatchSnapshot();
  });
});
