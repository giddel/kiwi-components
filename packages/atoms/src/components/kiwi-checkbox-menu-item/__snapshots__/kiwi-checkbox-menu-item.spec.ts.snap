// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kiwi-checkbox-menu-item renders 1`] = `
<kiwi-checkbox-menu-item>
  <!---->
  <li class="checkbox-menu-item pointer">
    <span>
      Slot Child
    </span>
  </li>
</kiwi-checkbox-menu-item>
`;
